### PR TITLE
Automating-tasks, checking-repository-health and solving-the-problem

### PR DESCRIPTION
**Pull request by imisi-akande**
#### What does this PR do?
This PR contains outputs for automating-tasks, checking-repository-health, solving-the-problem, 
#### Description of Task to be completed?
1. A re-screencast showing gulp watching and refreshing your browser when changes are made to my HTML, CSS & JS files
2.  A re-screenshot of my code coverage and GPA badges in your repo's README file
3. The file containing the implementation including all relevant dependencies(validateFile method added)
#### How should this be manually tested?
Navigate to the output folder to view the solutions for each output
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
N/A
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: 06-building-a-program-start-to-finish/automating-tasks/Output/README.MD [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#### A screencast showing gulp watching and refreshing your browser.
+### Link to screencast
+[link to screen cast of gulp automation](http://youtu.be/zcBIFShQHNw?hd=1)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/automating-tasks/Output/README.MD:3]: This is what is expected for this output - _**a screencast showing gulp watching and refreshing your browser when changes are made to your HTML, CSS & JS files**_

What you have done only shows changes to your CSS file. Please redo video.

File: 06-building-a-program-start-to-finish/checking-repository-health/Output/README.MD [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# invertedindex
+
+#### Checking repository health using Code Climate and Coveralls
+[link to screenshot for my code coverage and GPA badges] (https://github.com/andela-iakande/invertedindex/blob/staging/readme.md)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/checking-repository-health/Output/README.MD:4]: This is not a screenshot!

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/Output/README.MD [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#### Using TravisCI
+[Link to a successful TravisCI build of`development` branch](https://travis-ci.org/andela-iakande/invertedindex/builds/215751053)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/Output/README.MD:2]: Good

File: 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#### Output
+- **Using**: GitHub
+- **Deliver**: the repository for your proposed project and at least one `Pull Request`
+- **With**: *independently*
+
+#### Link to GitHub repo
+  https://github.com/andela-iakande/invertedindex/tree/development
+
+#### Author
+ andela-iakande/invertedindex
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md:10]: Cool

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndexTest.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
+/* Test Setup */
+const myInvertedIndex = new InvertedIndexClass();
+const book = require('./../books.json');
+const emptyBook = require('./../bookempty.json');
+const wrongBook = require('./../bookwrongformat.json');
+const notBook = require('./../notBook.json');
+const books = require('./../books3.json');
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndexTest.js:7]: A base filename should exactly match the name of its default export. [See link](https://github.com/airbnb/javascript#naming--filename-matches-export)

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndexTest.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    it('should return false when no result is found', () => {
+      expect(myInvertedIndex.searchIndex('impossibility'))
+        .toBeFalsy();
+    });
+    it('should return true when the result is found', () => {
+      expect(myInvertedIndex.searchIndex('impossibility'))
+        .toBeFalsy();
+    });
+  });
+});
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndexTest.js:79]: Your test does not provide for all the expected methods your [InvertedIndex class](https://docs.google.com/document/d/1Y1171vb09G-KhvHafgK219vxf9bhfgkUBA_CxbGPAGU/edit) should have.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    return this.indexTable[fileName];
+  }
+
+  /**
+   * getTokens
+   * Obtain an array of terms from a string
+   * @param {String} string get token from string
+   * @return {Object} An array of the generated token
+   */
+  static tokenize(string) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:56]: Is "**string**" a datatype of the parameter you are passing to this method? I think you can do better to name your parameters correctly.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+   * @returns {Object} A map of the search result
+   */
+  searchIndex(searchTerms, fileNames) {
+    const fileTitle = fileNames || Object.keys(this.files);
+    this.result = {};
+    const allSearchTerms =
+      InvertedIndexClass.tokenize(searchTerms);
+    fileTitle.forEach((currentFile) => {
+      allSearchTerms.forEach((term) => {
+        if (Object.hasOwnProperty.call(this.indexTable[currentFile], term)) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:79]: Please see the [Airbnb Style Guide](https://github.com/airbnb/javascript#objects--prototype-builtins) on how best to call Object.prototype methods directly

**Review by andela-moe — COMMENTED** [review]
Please address comments

File: 06-building-a-program-start-to-finish/automating-tasks/Output/README.MD (hunk 1 above)
Comment (imisi-akande) [reply 06-building-a-program-start-to-finish/automating-tasks/Output/README.MD:3]: I have done a re-screencast of the video showing gulp watching and refreshing on my browser

File: 06-building-a-program-start-to-finish/checking-repository-health/Output/README.MD (hunk 2 above)
Comment (imisi-akande) [reply 06-building-a-program-start-to-finish/checking-repository-health/Output/README.MD:4]: Yes, I have posted the screenshot

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndexTest.js (hunk 5 above)
Comment (imisi-akande) [reply 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndexTest.js:7]: I have corrected the base filename to tally with the default export name

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndexTest.js (hunk 6 above)
Comment (imisi-akande) [reply 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndexTest.js:79]: I have written the test for all expected methods in my invertedIndexClass

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js (hunk 7 above)
Comment (imisi-akande) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:56]: I have renamed the parameter i am passing to this method as inputData.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js (hunk 8 above)
Comment (imisi-akande) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:79]: I have called the object.prototype method appropriately using the  object.prototype.hasOwnProperty as described in the Airbnb style

**Pull request closed without merging**